language-support/ts/codegen/tests/ts/daml-ledger-fetch/src/index.ts

```diff
@@ -189,8 +189,10 @@ class Ledger {
    */
   async exercise<T extends object, C, R>(choice: Choice<T, C, R>, contractId: ContractId<T>, argument: C): Promise<[R , Event<object>[]]> {
     const payload = {
-      templateId: choice.template().templateId,
-      contractId,
+      reference: {
```

Comment:
Why? I think the old API was fine.

Reply:
because reference can be of `ContractLocator` type:

```
  sealed abstract class ContractLocator[+LfV] extends Product with Serializable

  final case class EnrichedContractKey[+LfV](
      templateId: TemplateId.OptionalPkg,
      key: LfV
  ) extends ContractLocator[LfV]

  final case class EnrichedContractId(
      templateId: Option[TemplateId.OptionalPkg],
      contractId: domain.ContractId
  ) extends ContractLocator[Nothing]
```

Reply:
you see it groups `templateId` with `key` and optional `templateId` with `contractId`. That is what I currently have in my domain model, which I am not going to flatten that is for sure... JSON can be flattened, but I think `reference` field kind of makes sense. Again up to you.

Reply:
plus I hate writing JSON formatters manually :), if we have a `reference` field, the formatter can be completely derived.